### PR TITLE
Fix aggressive downgrade

### PR DESCRIPTION
**Pull request by davidfowl**
- Only warn about the downgrade if the thing downgrading is accepted
- Fix the printing of version ranges (pretty print)

https://github.com/NuGet/Home/issues/948

/cc @pranavkm @emgarten @anurse

### CODE REVIEW

**Comment by pranavkm** [comment]
:shipit:

File: src/NuGet.Commands/RestoreCommand.cs [hunk 1]
```diff
@@ -97,7 +97,11 @@ public Task<RestoreResult> ExecuteAsync()
                     var downgraded = downgrade.DowngradedTo;
                     var downgradedBy = downgrade.DowngradedFrom;
 
-                    _log.LogWarning($"Detected package downgrade: {downgraded.Key.Name} from {downgraded.Key.VersionRange.MinVersion} to {downgradedBy.Key.VersionRange.MinVersion} {Environment.NewLine} {downgraded.GetPath()} {Environment.NewLine} {downgradedBy.GetPath()}");
+                    // It's only a downgrade if this dependency edge was chosen
+                    if (downgradedBy.Disposition == Disposition.Accepted)
+                    {
+                        _log.LogWarning($"Detected package downgrade: {downgraded.Key.Name} from {downgraded.Key.VersionRange.MinVersion} to {downgradedBy.Key.VersionRange.MinVersion} {Environment.NewLine} {downgraded.GetPath()} {Environment.NewLine} {downgradedBy.GetPath()}");
```
Comment (yishaigalatzer) [inline src/NuGet.Commands/RestoreCommand.cs:103]: Needs loc - I'll file a separate bug

Comment (yishaigalatzer) [inline src/NuGet.Commands/RestoreCommand.cs:103]: https://github.com/NuGet/Home/issues/954 - Moved to 3.2 (Post UWP release)

**Comment by emgarten** [comment]
:shipit:

**Pull request closed without merging**